src/afl-fuzz-init.c

```diff
@@ -985,6 +1035,83 @@ void perform_dry_run(afl_state_t *afl) {
 
   }
 
+  /* Now we remove all entries from the queue that have a duplicate trace map */
+
+  q = afl->queue;
+  struct queue_entry *p, *prev = NULL;
+  int                 duplicates = 0;
+
+restart_outer_cull_loop:
+
+  while (q) {
+
+    if (q->cal_failed || !q->exec_cksum) { goto next_entry; }
```

Comment:
so many gotos... hard to follow and may be problematic for the compiler

Reply:
two gotos, and IMHO easy to read. also works like a charm. not using goto will make this much more complex.